### ISSUE TITLE
dead-link: support reference-style markdown links

### ISSUE THREAD

**Issue by alifurkangokce** [alifurkangokce/driftlint#8]
`dead-link` (v0.12) verifies inline links — `[text](docs/deploy.md#staging)`. Reference-style links are not checked yet:

```md
See the [deploy guide][deploy].

[deploy]: docs/deploy.md#staging
```

**Where to look:** `src/extract.ts` → `extractLinks()`, then `src/checks/links.ts` needs no change if the definition's line number is reported.

**What good looks like:** definitions collected per file, usages resolved against them, unused/undefined labels ignored (that's markdownlint's job, not ours). Tests go in `test/v12.test.js`.

**Comment by LunaMeerkats** [comment]
I’m going to take this on with a focused regression-first change against `eb80e16`.

Plan:

- add `v12` coverage for reference definitions and later full/collapsed usages, with findings attributed to the definition line;
- resolve labels per file while leaving unused definitions and undefined labels alone;
- preserve the current exclusions for fences, ignore markers, external targets, and other non-repository links;
- run the focused build/test gate, then the broadest feasible `npm test` validation and report any baseline-only limitation exactly.

This work is being handled autonomously by OpenAI Codex under the account shown here. I’ll keep the change single-purpose and disclose the resulting validation evidence.
